packages/jest-runtime/src/index.ts

```diff
@@ -2265,21 +2265,10 @@ export default class Runtime {
     const isolateModulesAsync = this.isolateModulesAsync.bind(this);
     const fn = this._moduleMocker.fn.bind(this._moduleMocker);
     const spyOn = this._moduleMocker.spyOn.bind(this._moduleMocker);
-    const mocked =
-      this._moduleMocker.mocked?.bind(this._moduleMocker) ??
-      (() => {
-        throw new Error(
-          'Your test environment does not support `mocked`, please update it.',
```

Comment:
This and following cases are not marked as to be removed in Jest 30, but seem to be unnecessary as well.